src/common/IsolatedWorld.ts

```diff
@@ -140,10 +139,11 @@ export class IsolatedWorld {
     return `${name}_${contextId}`;
   };
 
-  constructor(frame: Frame) {
+  constructor(frame: Frame, injected = false) {
```

Comment:
WDYT about creating `PuppeteerWorld extends IsolatedWorld` and extending the setContext to schedule evaluation of the injected scripts? it'd also help avoid boolean parameters on the constructor

Reply:
I'm not a big fan of extending a class as complicated as IsolatedWorld. In any case, we can do refactor this later.